environ/process/eigen_cluster/prepare_eigen_cluster.py

```diff
@@ -0,0 +1,325 @@
+warnings.filterwarnings("ignore")
+
+
+def _load_in_data_lst(
+    file_root: str,
+    filter_name: str = "",
+) -> list:
+    """
+    Function to generate a list of dataframes from a given path
+    """
+
+    # get the list of files
+    path = file_root
+    file_lst = glob.glob(path + "/*.csv")
+
+    # isolate the file with specific version
+    file_name_lst = (
+        [file_name for file_name in file_lst if filter_name == file_name.split("_")[-2]]
+        if filter_name
+        else file_lst
+    )
+
+    return file_name_lst
+
+
+def _generate_node_edge(
+    file_name: str,
+    edge_col: list[str],
+    weight_col: list[str],
+    exclude_special_route: bool = True,
+    dict2str: bool = False,
+    aggreate_weight: bool = False,
+    convert_undirected: bool = False,
+) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Function to generate the node and edge arrays
+    """
+
+    # load the data
+    df_network = pd.read_csv(file_name)
+
+    if exclude_special_route:
+        # Exclude "LOOP" AND "SPOON", and "Error" in intermediary
+        df_network = df_network[
+            (df_network["label"] == "0") & (df_network["intermediary"] != "Error")
+        ].copy()
+
+    if dict2str:
+        # unwrap the dictionary {'symbol': 'DAI'} by extracting the
+        # value after ': '' and before ''}' using split
+        for col in edge_col:
+            df_network[col] = df_network[col].apply(
+                lambda x: x.split(": '")[1].split("'}")[0]
+            )
+
+    # convert the weight to float
+    df_network[weight_col] = df_network[weight_col].astype(float)
+
+    if aggreate_weight:
+        # sum up dathe weight
+        df_network = df_network.groupby(edge_col)[weight_col].sum().reset_index()
+
+    if convert_undirected:
+        # convert the edge to undirected
+        df_network["edge"] = df_network[edge_col].apply(
+            lambda x: tuple(sorted(x)), axis=1
+        )
+        df_network = df_network.groupby("edge")[weight_col].sum().reset_index()
+        df_network[edge_col[0]] = df_network["edge"].apply(lambda x: x[0])
+        df_network[edge_col[1]] = df_network["edge"].apply(lambda x: x[1])
+
+    # get the edge list
+    edge = df_network[edge_col].to_numpy()
+
+    # get the weight list
+    weight = df_network[weight_col].to_numpy()
+
+    return edge, weight
+
+
+def _compute_indicator(
+    edge: np.ndarray,
+    weight: np.ndarray,
+    graph_type: Literal["directed_multi", "undirected_multi", "directed", "undirected"],
+    indicator_type: Literal["eigenvector", "clustering"] = "eigenvector",
+) -> pd.DataFrame:
+    """
+    compute indicators for a given graph
+    :param edge: a numpy array of edges
+    :param weight: a numpy array of weights
+    :param graph_type: a string to indicate the type of graph
+    :param indicator_type: a string to indicate the type of indicator
+    :return: a dataframe of eigenvector centrality
+    """
+
+    if graph_type == "directed_multi":
```

Comment:
use `case`

https://towardsdatascience.com/switch-case-statements-are-coming-to-python-d0caf7b2bfd3